google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -368,13 +368,107 @@ public void defaultQuery() throws Exception {
     assertEquals("bar", documents.next().get("foo"));
   }
 
+  @Test
+  public void defaultQueryStream() throws Exception {
+    addDocument("foo", "bar");
+    addDocument("foo", "bar");
+
+    final Semaphore semaphore = new Semaphore(0);
+    final Iterator<String> iterator = Arrays.asList("bar", "bar").iterator();
+    randomColl.stream(
+        new ApiStreamObserver<DocumentSnapshot>() {
+          @Override
+          public void onNext(DocumentSnapshot documentSnapshot) {
+            assertEquals(iterator.next(), documentSnapshot.get("foo"));
+          }
+
+          @Override
+          public void onError(Throwable throwable) {}
+
+          @Override
+          public void onCompleted() {
+            semaphore.release();
+          }
+        });
+
+    semaphore.acquire();
+    // Verify the number of response
+    assertFalse(iterator.hasNext());
```

Comment:
Assert document count instead of asserting  iterator emptiness would be clearer.

Reply:
`assertFalse(iterator.hasNext());` is also used in other existing test functions inside this file. IMO count is less intuitive than iterator since in Java test count needs to wrapped inside an array and introduces one more variable  to this test where the same purpose can be achieved by iterator.